iceoryx_posh/source/roudi/roudi_process.cpp

```diff
@@ -126,21 +134,154 @@ ProcessManager::ProcessManager(RouDiMemoryInterface& roudiMemoryInterface,
+                              << "' is still running after SIGKILL was sent " << processKillDelay.seconds<int>()
+                              << " seconds ago. RouDi is ignoring this process.";
+                }
+                ++i;
+            }
+        }
+    }
+
+    auto it = m_processList.begin();
+    while (removeProcess(it))
+    {
+        it = m_processList.begin();
+    }
+}
+
+bool ProcessManager::requestShutdownOfProcess(const RouDiProcess& process,
+                                              ShutdownPolicy shutdownPolicy,
+                                              ShudownLog shudownLog) noexcept
+{
+    static constexpr int32_t ERROR_CODE = -1;
+    auto killC = iox::cxx::makeSmartC(kill,
+                                      iox::cxx::ReturnMode::PRE_DEFINED_ERROR_CODE,
+                                      {ERROR_CODE},
+                                      {},
+                                      static_cast<pid_t>(process.getPid()),
+                                      (shutdownPolicy == ShutdownPolicy::SIG_KILL ? SIGKILL : SIGTERM));
+    if (killC.hasErrors())
+    {
+        if (shudownLog == ShudownLog::FULL)
+        {
+            switch (killC.getErrNum())
+            {
+            case EINVAL:
+                LogWarn() << "Process ID " << process.getPid() << " named '" << process.getName()
+                          << "' could not be killed with "
+                          << (shutdownPolicy == ShutdownPolicy::SIG_KILL ? "SIGKILL" : "SIGTERM")
+                          << ", because the signal sent was invalid.";
+                break;
+            case EPERM:
+                LogWarn() << "Process ID " << process.getPid() << " named '" << process.getName()
+                          << "' could not be killed with "
+                          << (shutdownPolicy == ShutdownPolicy::SIG_KILL ? "SIGKILL" : "SIGTERM")
+                          << ", because RouDi doesn't have the permission to send the signal to the target processes.";
+                break;
+            case ESRCH:
+                LogWarn() << "Process ID " << process.getPid() << " named '" << process.getName()
+                          << "' could not be killed with "
+                          << (shutdownPolicy == ShutdownPolicy::SIG_KILL ? "SIGKILL" : "SIGTERM")
+                          << ", because the target process or process group does not exist.";
+                break;
+            default:
+                LogWarn() << "Process ID " << process.getPid() << " named '" << process.getName()
+                          << "' could not be killed with"
+                          << (shutdownPolicy == ShutdownPolicy::SIG_KILL ? "SIGKILL" : "SIGTERM")
+                          << " for unknown reason: ’" << killC.getErrorString() << "'";
+            }
+        }
+        return false;
     }
+    return true;
+}
+
+bool ProcessManager::isProcessTerminated(const RouDiProcess& process) noexcept
```

Comment:
This is naming is weird. This method does not verify if a process is terminated it kills a process inside of `requestShutdownOfProcess`. I would get rid of this method and use `requestShutdownOfProcess` directly.

Reply:
That's right. Removed.